api/agent/pure_runner.go

```diff
@@ -202,24 +213,24 @@ type pureRunnerCapacityManager struct {
 
 type capacityDeallocator func()
 
-func newPureRunnerCapacityManager(units uint64) pureRunnerCapacityManager {
-	return pureRunnerCapacityManager{
+func newPureRunnerCapacityManager(units uint64) *pureRunnerCapacityManager {
+	return &pureRunnerCapacityManager{
 		totalCapacityUnits:     units,
 		committedCapacityUnits: 0,
 	}
 }
 
-func (prcm *pureRunnerCapacityManager) checkAndReserveCapacity(units uint64) error {
+func (prcm *pureRunnerCapacityManager) CheckAndReserveCapacity(units uint64) error {
 	prcm.mtx.Lock()
 	defer prcm.mtx.Unlock()
-	if prcm.committedCapacityUnits+units < prcm.totalCapacityUnits {
+	if prcm.committedCapacityUnits+units <= prcm.totalCapacityUnits {
```

Comment:
very minor... not likely with sane capacity numbers, but I'd revert the math here to avoid any overflow. eg;

```
if total - committed >= units {
```